src/models/attendance.js

```diff
@@ -8,5 +14,89 @@ export default function(sequelize, DataTypes) {
         freezeTableName: true
     });
 
+    details.uploadAttendance = function(req, res) {
+            return new Promise((resolve, reject) => {
+                var oldpath = config.path;
+                var newpath = path.join(__dirname, './uploads/AGL_001.TXT');
+                fs.rename(oldpath, newpath, function(err) { //stores file to new path in upload folder
+                    if (err) reject(err);
+                    else {
+                        var lineReader = require('readline').createInterface({
+                            input: fs.createReadStream(newpath) //read file from new path in upload folder
+                        });
+                        var count = 0;
+                        var line_array = [];
+                        var filtered_data = [];
+
+                        function readFile(callback) {
+                            lineReader.on('line', function(line) {
```

Comment:
divide this function in smaller functions. and create helper classes. if not clear ask